omod-1.8/src/test/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_8/UserController1_8Test.java

```diff
@@ -245,6 +313,28 @@ public void shouldFindUserByUsername() throws Exception {
 	}
 	
 	@Test
+	public void shouldFindUserByUsernameUsingRequiredSearchParameterForMockedHandler() throws Exception {
+		SimpleObject response = deserialize(handle(newGetRequest(getURI(), new Parameter("username", "admin"))));
+		List<Object> results = Util.getResultsList(response);
+		
+		Object next = results.iterator().next();
+		
+		assertEquals(1, results.size());
+		assertEquals(MOCKED_USER_NAME_FOR_REQUIRED_PARAM, PropertyUtils.getProperty(next, "display"));
+	}
+	
+	@Test
+	public void shouldFindUserByUsernameUsingOptionalSearchParameterForMockedHandler() throws Exception {
+		SimpleObject response = deserialize(handle(newGetRequest(getURI(), new Parameter("username", "bruno"))));
+		List<Object> results = Util.getResultsList(response);
+		
+		Object next = results.iterator().next();
+		
+		assertEquals(1, results.size());
+		assertEquals(MOCKED_USER_NAME_FOR_OPTIONAL_PARAM, PropertyUtils.getProperty(next, "display"));
+	}
```

Comment:
Some ideas for tests:
Define handlers:
1) required username:admin, optional preferredLocales
2) required systemId, optional username:bruno
3) optional username:bruno

1. Match 1) for query "?username=admin&preferredLocales=en"
2. Match 2) for query "?systemId=bruno"
3. Match 2) for query "?systemId=bruno&username=bruno"
4. Match 3) for query "?username=bruno"
5. Do not match 1) for "?username=bruno&preferredLocales=en"
6. Do not match 3) for "?username=james"
7. Do not match 2) for "?username=james&systemId=bruno"